checks.d/docker_daemon.py

```diff
@@ -479,14 +499,34 @@ def _report_container_size(self, containers_by_id):
             tags = self._get_tags(container, PERFORMANCE)
             m_func = FUNC_MAP[GAUGE][self.use_histogram]
             if "SizeRw" in container:
-
                 m_func(self, 'docker.container.size_rw', container['SizeRw'],
                        tags=tags)
             if "SizeRootFs" in container:
                 m_func(
                     self, 'docker.container.size_rootfs', container['SizeRootFs'],
                     tags=tags)
 
+    def _report_container_count_by_state(self, containers_by_id, state="Any"):
+        count = 0
+        tags = {}
+        filterlambda = lambda x: not self._is_container_excluded(x) and state is "Any" or container["State"] is state
+        filtered = list(filter(filterlambda, containers_by_id))
+        tags = self._get_tags(filtered[0], PERFORMANCE)
+
+        m_func = FUNC_MAP[GAUGE][self.use_histogram]
+        # Report docker.container.count if state is "Any", otherwise
+        # report docker.container.state_STATE.count
+        suffix = ".state_{}".format(state.lower()) if state is not "Any" else ""
```

Comment:
Hi @parkr 
The way we usually do this is to report `docker.containers.count` several times, each time with a different state tag. This could look like:

```python
# const
STATES = ['created', 'restarting', 'running', 'paused', 'exited', 'dead']
...
def _report_container_count_by_state(self, containers_by_id):
    tags = self._get_tags(filtered[0], PERFORMANCE)
    m_func = FUNC_MAP[GAUGE][self.use_histogram]
    for state in STATES:
        # the filter changes for each state
        filterlambda = lambda x: not self._is_container_excluded(x) and container["State"] is state
        filtered = list(filter(filterlambda, containers_by_id))
        # add the state tag
        tags.append('state:{}'.format(state)
        m_func(self, 'docker.container.count'.format(suffix), len(filtered), tags=tags)
```
This way you can sum the reported values of `docker.container.count` on a graph to get the total count, or split the graph over the `state` tag and have the detailed count.

Let me know if it's still unclear.

(this snippet is not tested and could be optimized)